ext/ruby_prof/rp_thread.c

```diff
@@ -20,7 +20,7 @@ You cannot create an instance of RubyProf::Thread, instead you access it from a
 #include "rp_profile.h"
 
 VALUE cRpThread;
-
+st_table* merge_method_table;
```

Comment:
This looks like global state to me, won't this conflict with multiple threads? 
If you need access to it, add a method to rp_thread.h:

```
st_table* thread_get_method_table(thread_data_t* thread);

Reply:
I have modified the methods now to pass the information from thread side into the call tree side and store it as a struct in the call_tree.c. I cannot do the thread.h becasue the call_tree is not having thread info and the struct is needed to call the merge_children code in rb_st_foreach